atc/scheduler/algorithm/group_resolver.go

```diff
@@ -61,26 +61,6 @@ func (r *groupResolver) Resolve(ctx context.Context) (map[string]*versionCandida
 	})
 	defer span.End()
 
-	for i, cfg := range r.inputConfigs {
-		if cfg.PinnedVersion == nil {
-			continue
-		}
-
-		version, found, err := r.vdb.FindVersionOfResource(ctx, cfg.ResourceID, cfg.PinnedVersion)
-		if err != nil {
-			tracing.End(span, err)
-			return nil, "", err
-		}
-
-		if !found {
-			notFoundErr := db.PinnedVersionNotFound{PinnedVersion: cfg.PinnedVersion}
-			span.SetStatus(codes.InvalidArgument)
-			return nil, notFoundErr.String(), nil
-		}
-
-		r.pins[i] = version
```

Comment:
As the `pins` field in `groupResolver` is no longer being explicitly used does it make sense to remove it from the struct entirely? It would then also not need to be set on line 47.